metpy/plots/station_plot.py

```diff
@@ -59,6 +61,10 @@ def __init__(self, ax, x, y, fontsize=10, spacing=None, transform=None, **kwargs
             for all plot types, such as `clip_on`.
 
         """
+        # Handle datetime indices as independent variables
+        if isinstance(x, (pd.DatetimeIndex, pd.TimedeltaIndex)):
```

Comment:
This is ok if we *have* to handle it here. I'd like to handle it more in a matplotlib-like way. Where does the error come out of if we don't do this?

Reply:
It's a "TypeError: invalid type promotion":

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-8de7bfcbd1f2> in <module>
     31 #x= dates.date2num(ts.index)   # works!  ... Thank you Ryan!
     32 stationplot = StationPlot(ax, x , y, fontsize=12)
---> 33 stationplot.plot_symbol('C', ts , sky_cover)
     34 
     35 ax.xaxis.set_major_locator(mdates.DayLocator())

~/git_repos/MetPy/metpy/plots/station_plot.py in plot_symbol(self, location, codes, symbol_mapper, **kwargs)
    162         # Make sure we use our font for symbols
    163         kwargs['fontproperties'] = wx_symbol_font.copy()
--> 164         return self.plot_parameter(location, codes, symbol_mapper, **kwargs)
    165 
    166     def plot_parameter(self, location, parameter, formatter='.0f', **kwargs):

~/git_repos/MetPy/metpy/plots/station_plot.py in plot_parameter(self, location, parameter, formatter, **kwargs)
    198         """
    199         text = self._to_string_list(parameter, formatter)
--> 200         return self.plot_text(location, text, **kwargs)
    201 
    202     def plot_text(self, location, text, **kwargs):

~/git_repos/MetPy/metpy/plots/station_plot.py in plot_text(self, location, text, **kwargs)
    233         text_collection = self.ax.scattertext(self.x, self.y, text, loc=location,
    234                                               size=kwargs.pop('fontsize', self.fontsize),
--> 235                                               **kwargs)
    236         if location in self.items:
    237             self.items[location].remove()

~/git_repos/MetPy/metpy/plots/_mpl.py in scattertext(self, x, y, texts, loc, **kw)
    213         # Add it to the axes and update range
    214         self.add_artist(text_obj)
--> 215         self.update_datalim(text_obj.get_datalim(self.transData))
    216         self.autoscale_view()
    217         return text_obj

~/git_repos/MetPy/metpy/plots/_mpl.py in get_datalim(self, transData)
    274             """
    275             full_transform = self.get_transform() - transData
--> 276             XY = full_transform.transform(np.vstack((self.x, self.y)).T)  # noqa: N806
    277             bbox = transforms.Bbox.null()
    278             bbox.update_from_data_xy(XY, ignore=True)

~/miniconda3/envs/devel/lib/python3.7/site-packages/numpy/core/shape_base.py in vstack(tup)
    281     """
    282     _warn_for_nonsequence(tup)
--> 283     return _nx.concatenate([atleast_2d(_m) for _m in tup], 0)
    284 
    285 

TypeError: invalid type promotion
```

Reply:
Hrmm...need to look at matplotlib's unit-handling code for `ax.text()` or `Text`. I don't think they need anything special to handle these Pandas classes, but we might need to borrow some updated code for our `TextCollection`.

Reply:
Got it. Just copied some of the unit conversion code from the `draw` method.